checks/raw/pinned_dependencies.go

```diff
@@ -498,3 +494,28 @@ var validateGitHubActionWorkflow fileparser.DoWhileTrueOnFileContent = func(
 
 	return true, nil
 }
+
+func isActionDependencyPinned(actionUses string) bool {
+	// Though GitHub Actions also allows actions to be specified as local filepaths
+	// (e.g. uses: <path/to/file/in/repo>), it's not yet practical to determine whether an
+	// arbitrary file that lives in the action context is a trustworthy, pinned workflow file
+	// So, those actions are treated as 'unpinned'
```

Comment:
I would argue pinning dependencies isn't about trust, it's about fixing dependencies you don't control. I'm not sure I would treat them as unpinned, although they would need to be analyzed for unpinned actions of their own?

Reply:
True, this comment is mostly referring to #2189. My hypothesis is that anything the action's local context can be used by `uses`, i.e. a step could write an arbitrary file and then the next step would use that. I'm just leaving a comment to revisit this behavior, this PR doesn't change how we handle those paths.

Reply:
I think we can treat local actions as pinned, since they are "vendored". I think https://github.com/ossf/scorecard/issues/2189 can be addressed by parsing the action.yml of the local action and looking for pinned dependencies in the yaml file.

Reply:
But isn't it possible for an action to modify or overwrite other vendored actions, effectively unpinning them?

Reply:
anytime you run an Action, it overwrites the $GITHUB_WORKSPACE directory, so yes. But that's how Actions work and their code is only available for the duration of the call.
```yaml
uses: checkout
uses: ./my/local/action
uses: some/external-action // This action overwrites the previous code
```

The goal of pinning Action is to ensure they don't change when the dependency has a new release, unless the user LGTM the changes. It also helps for inventory: you know exactly what code you ran (assuming the dependency also pins its dependency.. which may not be true :/). The code *may* be overwritten by malicious code, but that's a harder problem. At least we know what was the code of the direct dependency is .. ideally we'd carry on till the last further-away indirect dependency

Reply:
Gotcha. Yes, it's hard to capture all the ways that the local context may change before someone uses a local action, but it's a really common pattern, especially with `.github/workflows` and `.github/actions` - so we should allow it for now. But I think there's another improvement here, which is that if someone uses the `actions/checkout` action that's bundled with GitHub, we should make sure that the checkout is pinned to a SHA.